lib/std/time.zig

```diff
@@ -59,6 +59,395 @@ pub fn sleep(nanoseconds: u64) void {
     std.os.nanosleep(s, ns);
 }
 
+pub const TimeZone = @import("time/TimeZone.zig");
+
+pub const LocalTimeError = error{
+    /// The local time zone of the system could not be determined.
+    LocalTimeUnavailable,
+} || std.mem.Allocator.Error;
+
+pub fn localtime(allocator: std.mem.Allocator) LocalTimeError!TimeZone {
+    switch (builtin.os.tag) {
+        .linux => {
+            var tz_file = std.fs.openFileAbsolute("/etc/localtime", .{}) catch
+                return error.LocalTimeUnavailable;
+            defer tz_file.close();
+
+            var buffered = std.io.bufferedReader(tz_file.reader());
+            return TimeZone.parse(allocator, buffered.reader()) catch
+                return error.LocalTimeUnavailable;
+        },
+        else => @compileError("`localtime` has not been implemented for this OS!"),
+    }
+}
+
+/// ISO 8601 compliant date and time representation.
+pub const DateTime = struct {
+    /// nanoseconds, range 0 to 999
+    nanosecond: u16,
+
+    /// microseconds, range 0 to 999
+    microsecond: u16,
+
+    /// milliseconds, range 0 to 999
+    millisecond: u16,
+
+    /// seconds, range 0 to 60
+    second: u8,
+
+    /// minutes, range 0 to 59
+    minute: u8,
+
+    /// hours, range 0 to 23
+    hour: u8,
+
+    /// day of the month, range 1 to 31
+    day: u8,
+
+    /// day of the year, range 1 to 366
+    year_day: u16,
+
+    /// day of the week, enum Monday to Sunday
+    week_day: Weekday,
+
+    /// month, enum January to December
+    month: Month,
+
+    /// year, year 0000 is equal to 1 BCE
+    year: i32,
+
+    /// UNIX timestamp in nanoseconds
+    nano_timestamp: i128,
+    tt: TimeZone.TimeType,
```

Comment:
what is the implication of "TimeType" here? Does this cover a rule associated with the TimeZone (POSIX or transition lookup from the tzfile)?

Reply:
I don't like how this is currently either. I'll adjust it so that it is just the offset to the UNIX timestamp and a time zone name.

Reply:
I think @leroycep is currently working on an abstraction for this in his zig-chrono library. Basically, I think you'd like to have an interface type that could be a fixed offset, a POSIX TZ rule or a IANA db tzfile time zone as concrete type. Any concrete type would have a method that gives you the offset from UTC (+ a DST indicator if appropriate) for a given datetime, tz and Unix timestamp.

Reply:
Currently I have it organized like this:

```
tz.zig
tz/
    TZif.zig
     Posix.zig
```

`tz/TZif.zig` is the equivalent of the TimeZone file here, and `tz/Posix.zig` implements parsing and using POSIX TZ environment variable that is in the footer of most TZif files.

Like @FObersteiner mentioned, I have also been working on a TimeZone abstraction. [I decided to go with an `Allocator`-like interface](https://codeberg.org/geemili/chrono-zig/src/branch/master/src/tz.zig#L219-L285). Unfortunately this does make lifetimes of the TimeZones a bit more complex, which I'm handling at the moment by [having a `Database`](https://codeberg.org/geemili/chrono-zig/src/branch/master/src/tz.zig#L8-L190) which handles their allocation.